crates/forge-runner/src/trace_data.rs

```diff
@@ -26,310 +27,159 @@ pub const TRACE_DIR: &str = ".snfoundry_trace";
-            initial_gas,
-            contract_name,
-            function_name,
-        }
+#[must_use]
+pub fn build_profiler_execution_resources(value: ExecutionResources) -> ProfilerExecutionResources {
+    let mut syscall_counter = HashMap::new();
+    for (key, val) in value.syscall_counter {
+        syscall_counter.insert(build_profiler_deprecated_syscall_selector(key), val);
     }
-}
-
-#[derive(Clone, Copy, Debug, Default, Eq, Hash, PartialEq, Deserialize, Serialize)]
-pub enum ProfilerCallType {
-    #[default]
-    Call = 0,
-    Delegate = 1,
-}
-
-impl From<CallType> for ProfilerCallType {
-    fn from(value: CallType) -> Self {
-        match value {
-            CallType::Call => ProfilerCallType::Call,
-            CallType::Delegate => ProfilerCallType::Delegate,
-        }
+    ProfilerExecutionResources {
+        vm_resources: VmExecutionResources {
+            n_steps: value.vm_resources.n_steps,
+            n_memory_holes: value.vm_resources.n_memory_holes,
+            builtin_instance_counter: value.vm_resources.builtin_instance_counter,
+        },
+        syscall_counter,
     }
 }
 
-#[derive(Debug, Default, Clone, Deserialize, Serialize)]
-pub struct ProfilerExecutionResources {
-    pub vm_resources: VmExecutionResources,
-    pub syscall_counter: ProfilerSyscallCounter,
-}
-
-impl AddAssign<&ProfilerExecutionResources> for ProfilerExecutionResources {
-    fn add_assign(&mut self, rhs: &ProfilerExecutionResources) {
-        self.vm_resources += &rhs.vm_resources;
-        for (syscall, count) in &rhs.syscall_counter {
-            *self.syscall_counter.entry(*syscall).or_insert(0) += count;
-        }
+#[must_use]
+#[allow(clippy::needless_pass_by_value)]
+pub fn build_profiler_call_entry_point(
+    value: CallEntryPoint,
+    contracts_data: &ContractsData,
+) -> ProfilerCallEntryPoint {
+    let CallEntryPoint {
+        class_hash,
+        entry_point_type,
+        entry_point_selector,
+        storage_address,
+        call_type,
+        ..
+    } = value;
+
+    let mut contract_name = class_hash
+        .and_then(|c| contracts_data.class_hashes.get_by_right(&c))
+        .cloned();
+    let mut function_name = contracts_data.selectors.get(&entry_point_selector).cloned();
+
+    if entry_point_selector.0
+        == get_selector_from_name(TEST_ENTRY_POINT_SELECTOR)
+            .unwrap()
+            .into_()
+        && class_hash == Some(ClassHash(stark_felt!(TEST_CONTRACT_CLASS_HASH)))
+    {
+        contract_name = Some(String::from(TEST_CODE_CONTRACT_NAME));
+        function_name = Some(String::from(TEST_CODE_FUNCTION_NAME));
+    }
+    ProfilerCallEntryPoint {
+        entry_point_type: build_profiler_entry_point_type(entry_point_type),
+        entry_point_selector: EntryPointSelector(format!("{}", entry_point_selector.0)),
+        contract_address: ContractAddress(format!(
+            "0x{}",
+            hex::encode(storage_address.0.key().bytes())
+        )),
```

Comment:
```suggestion
        contract_address: ContractAddress(format!(
            "{}",
            storage_address.0.key()
        )),
```

and we can remove `hex` dependency

Reply:
Good point, just blindly copied it from the profiler code.